golang/src/permissions/restapi/impl/db/resources.go

```diff
@@ -127,35 +147,41 @@ func UpdateResource(tx *sql.Tx, id *string, name *string) (*models.ResourceOut,
 	row := tx.QueryRow(query, name, id)
 
 	// Return the result.
-	var resource models.ResourceOut
-	if err := row.Scan(&resource.ID, &resource.Name, &resource.ResourceType); err != nil {
-		return nil, err
-	}
-	return &resource, nil
+	return rowToResource(row)
 }
 
-func ListResources(tx *sql.Tx) ([]*models.ResourceOut, error) {
+func ListResources(tx *sql.Tx, resourceTypeName, resourceName *string) ([]*models.ResourceOut, error) {
 
 	// Query the database.
-	query := `SELECT r.id, r.name, t.name AS resource_type
+	var rows *sql.Rows
+	var err error
+	if resourceTypeName != nil && resourceName != nil {
```

Comment:
This is a longer-term comment (i.e. ignore me if you'd like!), but I'm mildly uncomfortable with the repetitiveness of these chains of else-ifs. I wonder what you'd think of using something like https://github.com/Masterminds/squirrel (much as I wonder at my own sanity suggesting a programmatic query generation library!) to enable us to not repeat the same `SELECT r.id, r.name, t.name AS resource_type` that's in each branch of the if here.